tests/flightplan/test_waypointstrategy.py

```diff
@@ -0,0 +1,167 @@
+from __future__ import annotations
+
+from pathlib import Path
+
+import pytest
+from dcs.terrain import Terrain, Caucasus
+from pytest import approx
+from shapely.geometry import Point, MultiPolygon
+
+from game.flightplan.waypointstrategy import WaypointStrategy, angle_between_points
+from game.utils import meters, Heading
+
+
+@pytest.fixture(name="terrain")
+def terrain_fixture() -> Terrain:
+    return Caucasus()
+
+
+def test_safe_prerequisite_safe_point() -> None:
+    strategy = WaypointStrategy(MultiPolygon([]))
+    strategy.prerequisite(Point(0, 0)).safe()
```

Comment:
Tiny comment but just something I noticed here.  Perhaps doing something like `to_be_safe()`, `is_safe()`, or`should_be_safe()` or similar instead of `safe()` would be a little more readable?  I wasn't completely certain what this was doing until I read the test `test_safe_prerequisite_unsafe_point()`. (Also thank you for writing tests!)

  I suppose we could also define multiple functions of the same functionality if you feel like there are different contexts that are suited for different ones.  Though that might be a detriment unless they could all live within the same file/area.

Reply:
Reading the other tests it may be worth changing other functions such `nearest` as well.  I read `test_no_solution_if_prerequisites_failed` as "Find a safe point nearest to {nearest Point} if {prerequisite Point} is safe."

(If I seem to be misunderstanding the premise here please let me know.)

Reply:
+1 to renaming `safe()` to `is_safe()`, done.

Reply:
For the second point, I added some comments:

```python
def test_no_solution_if_prerequisites_failed() -> None:
    """Verify that no solution is found if prerequisites are not satisfied.

    This test has a 1-meter radius threat zone about the center of the plane. It has a
    prerequisite for a safe center, which will fail. The test verifies that even if
    there are no .require() constraints that would prevent finding a solution, failed
    prerequisites still prevent it (prerequisites differ from constraints in that they
    will prevent any of the other operations from happening without needing to location
    constraints, which is important because it allows strategies to avoid defending
    against invalid cases).
    """
    strategy = WaypointStrategy(MultiPolygon([Point(0, 0).buffer(1)]))
    strategy.prerequisite(Point(0, 0)).is_safe()
    # This constraint won't actually apply, but it's required before calling find() so
    # we need to set it even though it's not actually relevant to the test.
    strategy.nearest(Point(0, 0))
    assert strategy.find() is None
```